block/src/Pos/Block/Verification.hs

```diff
@@ -0,0 +1,158 @@
+{-# LANGUAGE TypeFamilies  #-}
+{-# LANGUAGE TypeOperators #-}
+
+-- This module could be in core, but 'verifySscPayload' is kind of
+-- complex and I think it can't be moved there.
+
+-- | Pure verification for block. See "Pos.Util.Verification" for more info.
+
+module Pos.Block.Verification () where
```

Comment:
So everything is done by typeclass instances? What's the reason for this? What's the advantage over just using a value (verification function)?

Reply:
Advantage is that it's clear for you, which function to trigger to verify the data.